src/devicecontroller/DefaultDeviceController.ts

```diff
@@ -555,6 +555,10 @@ export default class DefaultDeviceController implements DeviceControllerBasedMed
           return (deviceMediaTrackConstraints.deviceId as ConstrainDOMStringParameters)
```

Comment:
https://github.com/aws/amazon-chime-sdk-js/blob/03a195883a5bd306bdbd1e87126e4070dc7bdc4e/src/devicecontroller/DefaultDeviceController.ts#L550

This line will fail if `deviceId` is a string because it's trying to access `exact` on the `string`.  Please test `this.getDeviceIdStr` with different types of inputs.

```
if (typeof deviceMediaTrackConstraints.deviceId === 'string') {
  return deviceMediaTrackConstraints.deviceId;
} else if ('exact' in deviceMediaTrackConstraints.deviceId) {
  return deviceMediaTrackConstraints.deviceId.exact;
}
```

Reply:
@kaaaren I'm looking into two methods `getDeviceIdStr` and `getActiveDeviceId` now and will share you with some suggestions today.

Reply:
The custom type ```Device``` can be `string | MediaTrackConstraints | MediaStream | null`. In case the `device` argument is `MediaTrackConstraints`, `device.deviceId` can be:
|Type|Handled by SDK?|
|--|--|
| a string | ✅ Added and handled by this PR |
| a string array | |
| an object with a string `exact` | ✅ |
| an object with a string array `exact` | |
| an object with a string `ideal` | |
| an object with a string array `ideal` | |

We are missing 4 other cases. In this PR, let's handle your string change. I'll create an issue for this.

```
private getDeviceIdStr(device: Device): string | null {
  if (device === null) {
    return null;
  } else if (typeof device === 'string') {
    return device;
  } else if ((device as MediaStream).id) {
    return (device as MediaStream).id;
  }

  const constraints: MediaTrackConstraints = device as MediaTrackConstraints;
  if (!constraints.deviceId) {
    return '';
  } else if (typeof constraints.deviceId === 'string') {
    return constraints.deviceId;
  }

  const deviceIdConstraint: ConstrainDOMStringParameters = constraints.deviceId as ConstrainDOMStringParameters;
  if (typeof deviceIdConstraint.exact === 'string') {
    return deviceIdConstraint.exact;
  }

  return '';
}
```


Need to update  `getActiveDeviceId` too

```
private getActiveDeviceId(kind: string): string | null {
  /* istanbul ignore else */
  if (this.activeDevices[kind] && this.activeDevices[kind].constraints) {
    const activeDeviceMediaTrackConstraints =
      this.activeDevices[kind].constraints.audio || this.activeDevices[kind].constraints.video;
    const activeDeviceConstrainDOMStringParameters = (activeDeviceMediaTrackConstraints as MediaTrackConstraints)
      .deviceId;

    let activeDeviceId: string;
    if (typeof activeDeviceConstrainDOMStringParameters === 'string') {
      activeDeviceId = activeDeviceConstrainDOMStringParameters as string;
    } else {
      activeDeviceId = (activeDeviceConstrainDOMStringParameters as ConstrainDOMStringParameters).exact as string;
    }

    return activeDeviceId;
  }
  /* istanbul ignore next */
  return null;
}
```

Reply:
Nice job Kyu. Very through comment 👍

Reply:
Thanks for the detailed explanation!! Will update

Reply:
Yeah while writing this snippet we took into consideration only the constrains types being used in our DefaultDeviceController in our SDK like the string, null and the one with the exact keyword. 
We use them in the proposed constraints for the parameter in getUserMedia()